plugins/obs-nvenc/obs-nvenc-test/obs-nvenc-test.cpp

```diff
@@ -47,9 +47,26 @@ typedef nvmlReturn_t (*NVML_GET_DEVICE_NAME)(nvmlDevice_t, char *, unsigned);
 typedef nvmlReturn_t (*NVML_GET_DEVICE_PCIE_GEN)(nvmlDevice_t, unsigned *);
 typedef nvmlReturn_t (*NVML_GET_DEVICE_PCIE_WIDTH)(nvmlDevice_t, unsigned *);
 typedef nvmlReturn_t (*NVML_GET_DEVICE_NAME)(nvmlDevice_t, char *, unsigned);
+typedef nvmlReturn_t (*NVML_GET_DEVICE_ARCHITECTURE)(nvmlDevice_t, unsigned *);
 typedef nvmlReturn_t (*NVML_GET_ENCODER_SESSIONS)(nvmlDevice_t, unsigned *, void *);
 typedef nvmlReturn_t (*NVML_GET_ENCODER_CAPACITY)(nvmlDevice_t, nvmlEncoderType, unsigned *);
 typedef nvmlReturn_t (*NVML_GET_ENCODER_UTILISATION)(nvmlDevice_t, unsigned *, unsigned *);
+
+/* Only Kepler is defined in NVIDIA's documentation, but it's also the main one we care about. */
+constexpr uint32_t NVML_DEVICE_ARCH_KEPLER = 2;
```

Comment:
I don't see `NVML_DEVICE_ARCH_KEPLER` being used anywhere else in this PR, why is it the only value that is expressly declared as a `constexpr`?

Reply:
It's used in the map below as the starting point, it's the one actually explicitly defined by NVIDIA so I carried that over here so it's clear where it came from: https://docs.nvidia.com/deploy/archive/R520/nvml-api/group__nvmlVgpuStructs.html#group__nvmlVgpuStructs_1g5c04580a96559779cc10538b57ec683c